src/ydb.cc

```diff
@@ -3295,9 +3298,9 @@ env_dbrename(DB_ENV *env, DB_TXN *txn, const char *fname, const char *dbname, co
             // otherwise, we're okay in marking this ft as remove on
             // commit. no new handles can open for this dictionary
             // because the txn has directory write locks on the dname
-            if (txn && !can_acquire_table_lock(env, txn, new_iname.get())) {
-                r = DB_LOCK_NOTGRANTED;
-            }
+            if (txn) {
+            	r = can_acquire_table_lock(env, txn, new_iname.get());
+	     }
```

Comment:
Sorry, I missed this the first time around, using tabs not spaces or combination of tabs and spaces. Should be spaces only.